crates/ra_hir/src/ty/traits.rs

```diff
@@ -42,7 +43,7 @@ pub(crate) fn impls_for_trait(
     }
     let crate_impl_blocks = db.impls_in_crate(krate);
     impls.extend(crate_impl_blocks.lookup_impl_blocks_for_trait(&trait_));
-    impls.into()
+    impls.into_iter().collect::<Vec<_>>().into()
```

Comment:
Hm, that makes me think: is it at all ok to use FxHashSets with salsa? I guess, we'll need to go out of our way to expose problems with non-determinism (we don't even run gc concurrently), so let's just use them until we hit a real problem :D